packages/web-components/src/components/global-header/components/global-header/src/globals/utils.ts

```diff
@@ -0,0 +1,133 @@
+  Home16,
+  IbmCloudKeyProtect16,
+  Launch16,
+  Login16,
+  Logout16,
+  Notification20,
+  RequestQuote16,
+  Settings16,
+  Share16,
+  User16,
+  User20,
+  User24,
+  User32,
+  UserFollow16,
+} from '@carbon/icons/es';
+import { toSVG, getAttributes } from '@carbon/icon-helpers';
+import { EventProps } from '../types/Header.types';
+
+export const ASSIST_ME_SCRIPT_DEV =
+  'https://ibmassistme-dev.zll1vg8lrcq.us-south.codeengine.appdomain.cloud/resources/assist-me/controller.js';
+export const ASSIST_ME_SCRIPT_PROD =
+  'https://www.ibm.com/resources/assist-me/controller.js';
+
+/**
+ *
+ * @param {string} iconType string value for the carbon icon
+ * @param {number} size number to indicate the size of the icon
+ * @param {string} slot string name of the slot that the icon will be displayed in
+ * @returns an svg of the icon if it is found in CarbonIcons
+ */
+export const renderCarbonIcon = (
+  iconType: string,
+  size: number,
+  slot?: string
+) => {
+  const iconWithSize = `${iconType}${size}`;
+  // console.log('iconWithSize', iconWithSize)
+
+  const CarbonIcons = {
+    AiLaunch20,
+    AppConnectivity20,
+    ArrowRight16,
+    Checkmark16,
+    Document16,
+    Email16,
+    Help20,
+    Home16,
+    IbmCloudKeyProtect16,
+    Launch16,
+    Login16,
+    Logout16,
+    Notification20,
+    RequestQuote16,
+    Settings16,
+    Share16,
+    User16,
+    User20,
+    User24,
+    User32,
+    UserFollow16,
+  };
+
+  const iconsRecord = CarbonIcons as Record<
+    string,
+    any /* eslint-disable-line @typescript-eslint/no-explicit-any */
+  >;
+  const LocalIcon = iconsRecord[iconWithSize];
+
+  // Check if the icon exists
+  if (!LocalIcon) {
+    console.error(`Icon "${iconWithSize}" not found in CarbonIcons.`);
+    return html`<div class="icon-container">Icon not found</div>`;
+  }
+
+  // Generate the SVG element with updated attributes
+  const svgElement: Element = toSVG({
+    ...LocalIcon,
+    attrs: { ...getAttributes(LocalIcon.attrs), slot },
+  });
+
+  // Serialize the SVG element to a string
+  // const svgString: string = new XMLSerializer().serializeToString(svgElement);
+  return svgElement;
+};
+
+export const isValidObject = (obj: unknown): obj is Record<string, unknown> => {
+  return (
+    obj !== null &&
+    typeof obj === 'object' &&
+    !Array.isArray(obj) &&
+    Object.keys(obj).length > 0
+  );
+};
+
+export function getAssistMeUrl(environment: string) {
+  switch (environment) {
+    case 'production':
+      return ASSIST_ME_SCRIPT_PROD;
+    default:
+      return ASSIST_ME_SCRIPT_DEV;
+  }
+}
+
+export const trackEvent = (eventName: string, props: EventProps) => {
+  const analytics =
+    /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+    (window as any)?.bluemixAnalytics;
+  if (analytics) {
+    props.platformTitle = 'IBM webMethods Hybrid Integration';
+    console.debug('Emit trackEvent', eventName, props);
+    analytics.trackEvent(eventName, props);
+  }
+};
```

Comment:
I'm assuming these utilities existed from some other repo, do you want to still include them in the labs component? I'm just thinking about if someone else uses this component, you don't want your instrumentation data to have invalid results.

Reply:
good point, I'll remove the platformTitle bit